### PR TITLE
KEYCLOAK-11678 Remove dummy resource. Adding keycloak-services and liquibase to jandex indexing

### PR DESCRIPTION
**Pull request by mposolda**
Summary of this PR changes:

- Removes the Dummy JAX-RS resource, which was before used as a workaround to register builtin RestEasy providers. By default, the Quarkus uses Jandex for indexing just the application classes and modules with the beans.xml. However when using the properties "quarkus.index-dependency" as described in the docs https://quarkus.io/guides/cdi-reference#how-to-generate-a-jandex-index, it can be used to register resources inside org.keycloak:keycloak-services module, which contains most of Keycloak JAX-RS resources and it is sufficient to enforce RestEasy to register all the required built-in providers.

- The @Provider annotations were removed from ObjectMapperResolver and KeycloakErrorHandles as those classes are registered anyway in the KeycloakApplication. Checked that it works on Keycloak on Undertow, Wildfly and Quarkus without any issues.

- In this PR, I've enabled Jandex indexing also for the Liquibase module. This allows to remove workaround in the pom.xml with unpacking/indexing liquibase and then manually read the index in LiquibaseProcessor. Instead we can rely on quarkus with reading the index.

@pedroigor @stianst WDYT? Do you eventually prefer to have resteasy and liquibase changes in separate PRs?

### CODE REVIEW

**Review by stianst — COMMENTED** [review]
LGTM @pedroigor should take a look as well before merging though

**Review by pedroigor — APPROVED** [review]
LGTM. @mposolda I'm also removing Dummy here https://github.com/keycloak/keycloak/pull/7290 and for this one, we don't actually need to index the services to be able to remove it. The reason being that we have the `QuarkusWelcomeResource`.

When building the application the JAX-RS classes from deps are already indexed at build time. 

Just a minor consideration but I'm approving.

**Comment by mposolda** [comment]
@pedroigor Thanks for the approve. Actually that was my thinking as well (regarding QuarkusWelcomeResource). However QuarkusWelcomeResource is not sufficient due the algorithm in ResteasyCommonProcessor.collectDeclaredProviders . With QuarkusWelcomeResource is the ResteasyDeployment.registerBuiltins set as false and some "builtin" Resteasy providers are not registered.  Which is a problem as some functionalities won't work.

**Pull request merged**
